vertx-web-client/src/test/java/io/vertx/ext/web/client/WebClientTest.java

```diff
@@ -1056,7 +1056,28 @@ public void testTLSDisabledEnableRequestTLSAbsURI() throws Exception {
     testTLS(false, true, client -> client.getAbs("https://" + DEFAULT_HTTPS_HOST + ":" + DEFAULT_HTTPS_PORT));
   }
 
+  /**
+   * Regression test for issue #563 (https://github.com/vert-x3/vertx-web/issues/563)
+   * <p>
+   * Only occurred when {@link WebClientOptions#isSsl()} was false for an SSL request.
+   */
+  @Test
+  public void testTLSQueryParametersIssue563() throws Exception {
+    testTLS(false, true,
+      client -> client.getAbs("https://" + DEFAULT_HTTPS_HOST + ":" + DEFAULT_HTTPS_PORT)
+        .addQueryParam("query1", "value1")
+        .addQueryParam("query2", "value2"),
+      serverRequest -> {
+        assertEquals("query1=value1&query2=value2", serverRequest.query());
+        return null;
+      });
+  }
+
   private void testTLS(boolean clientSSL, boolean serverSSL, Function<WebClient, HttpRequest<Buffer>> requestProvider) throws Exception {
+    testTLS(clientSSL, serverSSL, requestProvider, null);
+  }
+
+  private void testTLS(boolean clientSSL, boolean serverSSL, Function<WebClient, HttpRequest<Buffer>> requestProvider, Function<HttpServerRequest, Void> serverAssertions) throws Exception {
```

Comment:
Could you please change `Function<HttpServerRequest, Void> serverAssertions` to `Consumer<HttpServerRequest> serverAssertions` ?